### DISCUSSION TITLE
How to get my data as per my requested format from mongodb

### DISCUSSION THREAD

**Discussion by jyoti-ranjan-4** [prisma/prisma#23531] [Q&A]
### Question

Hello Team,
I've my data in my database in the below format:
```
{
  "cloud": "AZ",
  "time_stamp": "2023-10-17T07:16:31Z",
  "ABC-AZ-1": {
    "instance_name": "ABC-AZ-1",
    "instance_id": "abc/4332",
    "os": "Windows",
    "version": "18.04.202306070",
"business":"ABC"
  },
  "ABC-AX-223": {
    "instance_name": "ABC-AX-12",
    "instance_id": "axc/4fdsd32",
    "os": "Linux",
    "version": "18.04.202306070",
    "business":"xyz"
  }
}
```
And I've the schema like this:
```
model Instances {
  id         String @id @default(auto()) @map("_id") @db.ObjectId
  cloud      String
  time_stamp String
  instances  Json?
}
```

1. Can we collect data whose business name is "xyz" or any other based on the requirement?
If possible please suggest me the way.
2. I've a use case: In my application, if i onboard a new business, then i need to create a new collection and same in schema. Is there any way to create a schema automatically, once I create a business then the schema will create automatically?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @jyoti-ranjan-4 👋 

Thank you for raising this question.

> Can we collect data whose business name is "xyz" or any other based on the requirement? If possible please suggest me the way.

Do you mean you want to query the data based on the business field in your JSON object? If yes, then this possible. Here is an example of how you could query for instances where business is `xyz`. See this [section](https://www.prisma.io/docs/orm/prisma-client/special-fields-and-types/working-with-json-fields#filter-on-a-json-field) of the documentation.
```js
const instances = await prisma.instances.findMany({
  where: {
    instances: {
      path: ["ABC-AX-223", "business"],
      equals: "xyz",
    },
  },
});
```
This will return all instances where the `business` field of the `ABC-AX-223` object is equal to `xyz`.


> I've a use case: In my application, if i onboard a new business, then i need to create a new collection and same in schema. Is there any way to create a schema automatically, once I create a business then the schema will create automatically?

Prisma does not currently support automatic schema creation based on changes in your data. The Prisma schema is a representation of your database schema and needs to be manually updated to reflect changes in your database. However, if you're dealing with a dynamic nature of data, one approach is to use Prisma's `Json` data type field inside the schema and store any arbitrary data deeply nested inside the said field. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by jyoti-ranjan-4** [comment]
So let me discuss the use case properly. So we have different businesses, and for each business, we've got their respective collections. If I will add one more business then we need to write one more model in schema.prisma. If my application is running, and i'm creating a new business then i need to create a new model in schema.prisma. Is this possible?

**Reply by ludralph** [reply]
Prisma does not currently support creating models dynamically at runtime. This means that if you need to create a new model in your Prisma schema when you add a new business, you would need to manually update your Prisma schema and then run `prisma generate` to update your Prisma Client.

There is an open [issue](https://github.com/prisma/prisma/issues/21415) discussing the possibility of creating models at runtime but as of now, this feature is not available. If you have a large number of businesses and each one requires its own model, you might want to consider a different approach. For example, you could have a single Business model with a Json field to store the unique data for each business. This would allow you to add new businesses without needing to update your Prisma schema each time.

**Reply by jyoti-ranjan-4** [reply]
So my use case says:
Create different new models on the creation of new businesses. We've to get the data based on the business dynamically. Let's say if the user's business is Finance, then it will access the finance model. If the user belongs to Production business, then it will access the production model mentioned in my prisma.schema. As you said, we can't create a model dynamically so it's not possible to do that. If any other way is there, then please suggest me.

**Reply by ludralph** [reply]
Hi @jyoti-ranjan-4 

This is currently not possible using Prisma. You may want to open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your usecase so that our engineers may take a look.

**Comment by jyoti-ranjan-4** [comment]
So my use case says: 
Create different new models on the creation of new businesses. We've to get the data based on the business dynamically. Let's say if the user's business is Finance, then it will access the finance model. If the user belongs to Production business, then it will access the production model mentioned in my prisma.schema. As you said, we can't create a model dynamically so it's not possible to do that. If any other way is there, then please suggest me.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
